### ISSUE TITLE
How is a E/E System Architecture manageable in a iterative Process like  in SCRUM

### ISSUE THREAD

**Issue by marcusvw** [localmotors/veh-engineering#6]
We are fighting here with the Problem that a EE System Architecture includes a lot of interconnected items, regarding functional relations, safety relations, function assignments related to HW and SW etc.  We need within the SCRUM cycle some steps to define the consistency and also the efficiency of the E/E System Architecture.
A "classical" / linear approach is the PreeVision Work Flow:
https://media.licdn.com/mpr/mpr/shrinknp_1000_1000/AAEAAQAAAAAAAAaFAAAAJGJjNjkzMjIyLWFjMzItNDZhNS04OTNmLTlmMDA5YzRkMjlhOA.png

Especially the first three steps are problematic as they are not in the typical SCRUM Cycle and having  complex relations. 
Requirements-> Logical Architecture-> SW Architecture / Function distribution 

SW Development, if the SW-Architecture exists can be projected to a standard SCRUM cycle.

**Comment by creighto** [comment]
Actually, the problems with the first three steps have been studied quite a bit, too. I would suggest that we follow the so called "twin peaks" model: http://dx.doi.org/10.1109/2.910904

In a nutshell, I would see three "gears" in our Scrum-of-Scrums synchronization scheme:

## Rough Architectural Layers and Their Release Cycle Time
Layer                                      | Release Cycle Times
----------------------------|----------------------
Requirements+Architecture | Medium
Software                                | Short
Hardware                               | Long

**Comment by marcusvw** [comment]
In general that was also the point I'm following. 
The point is actually that we will need a  Release of changes/new Requirements, that is the reason why separated Requirements and Architecture. We can also do this by an added review and release activity in this cycle. In the actual proposed way the requirement must "survive" the Analysis and Design Cycle to get into the Architecture Cycle, this has the advantage that we have always clearly defined inputs for these cycles. Another point for the separation is that we can filter/have a defined decision process for the Co-Creation approach.

 I would differentiate regarding HW Development [ECU Development ] and [Harness Changes, adding COTS components]. The Harness and COTS components can be handled in your reference with the Software Cycle. In the case we have an adaption to an ECU Hardware (in the case that we have our "own" hardware) this would be the Hardware Cycle (I've actually not considered this, could be a discussion point). That was also the reason why I was not calling it Software but implementation Cycle.

**Comment by creighto** [comment]
I think it would be helpful if we narrowed the scope of this ticket to a more digestable size. Having had a "sneak peek" at your development branch, I think the pull request will become too big for a meaningful review.

Do we have any way of reducing the scope to a single architectural layer, for example?

**Comment by marcusvw** [comment]
Perhaps we should split the approach in a RE, System Design, System Implementation, Vehicle validation cycle<---(actually mentioned in the example document but not described)

**Comment by marcusvw** [comment]
Actually I'm thinking about one issue regarding doing safety concept in a cyclic way, if we have functions with ASIL C and above we will need an external accessor/review. Here we need a partner doing this always with the same person. If you have to understand all things again and review it again it would be quite difficult, on the other hand we need for the documentation a way that makes it quite easy to trace/trach changes.

**Comment by creighto** [comment]
Yes, let's split this issue up. How about so:

## Rough Architectural Layers and Their Release Cycle Time
Layer                                      | Release Cycle Times | Sub-Ticket Title | Comment |
----------------------------|----------------------|-----------------|-----------|
Requirements+Architecture | Medium                      | need agile process description for Requirements+Architecture | this cycle will create the function decomposition (and consequential SW/HW deployment) and create linked-back, new entries in the backlogs of HW+SW layers | 
Software                                | Short                          | need agile process description for Software | this cycle will actually be instantiated multiple times, depending on which hardware target the architecture determined | 
Hardware                               | Long                           | need agile process description for Hardware | this cycle will actually be instantiated multiple times, depending on which hardware units the architecture determined | 


In a nutshell, the HW+SW cycles are process classes to me. Both classes must contain the Design, Implementation, and Validation steps.

**Comment by creighto** [comment]
And about the external assessors: I think we should make it a general rule that we do like to keep responsibilities with the same people as much as we can. I think this is doable, especially, if we can keep our development environment open to external assessors like here.
